### PR TITLE
Add save comic functionality and minor bug fixing and exception handling messages

### PR DESCRIPTION
**Pull request by shaleenx**
Add save comic functionality and minor security related bug fixing like closing file descriptors. Also, the script has been made more verbose in incidences of exceptions.

### CODE REVIEW

File: xkcdcl [hunk 1]
```diff
@@ -51,10 +55,14 @@ class XKCDCL:
             # Fetch the data from API endpoint
             data = json.load(urllib2.urlopen(self.api_url))
         except:
-            return
+		print traceback.format_exc()
```
Comment (Somsubhra) [inline xkcdcl:58]: Indentation does not match the rest of code.

File: xkcdcl [hunk 2]
```diff
@@ -68,14 +76,20 @@ class XKCDCL:
...
             comic = Image.open(comic_file)
             comic.show()
+	    user = str(getlogin())
+	    comic.save("/home/"+user+"/Pictures/XKCD-Fetcher/"+str(self.comic_id)+".png")
+	    fd.close()
+
         except:
-            print "Error: The comic does not exist"
-            return
+		print traceback.format_exc()
```
Comment (Somsubhra) [inline xkcdcl:84]: Here also the indentation doesn't match

File: xkcdcl [hunk 3]
```diff
@@ -68,14 +76,20 @@ class XKCDCL:
             # Show the image using PIL
             comic = Image.open(comic_file)
             comic.show()
+	    user = str(getlogin())
+	    comic.save("/home/"+user+"/Pictures/XKCD-Fetcher/"+str(self.comic_id)+".png")
```
Comment (Somsubhra) [inline xkcdcl:80]: Use space between 'var + var'

File: xkcdcl (hunk 1 above)
Comment (shaleenx) [inline xkcdcl:58]: My bad. Looks like you have already corrected the code.

**Comment by Somsubhra** [comment]
Fixed all issues with coding conventions and directory creation.

**Pull request merged**